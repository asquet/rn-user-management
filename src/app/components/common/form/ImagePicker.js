import React from 'react';
import { View, Image, TouchableOpacity } from 'react-native';
import ImagePicker from 'react-native-image-picker';
import { Text } from '../content/Text';
import { formStyles } from '../../../stylesheet';

export default class MyImagePicker extends React.Component {

  constructor(props) {
    super(props);

    this.state = {
      avatarSource: null,
    };
  }

  showPicker() {
    // More info on all the options is below in the README...just some common use cases shown here
    const options = {
      title: 'Select Avatar',
      customButtons: [
        { name: 'fb', title: 'Choose Photo from Facebook' },
      ],
      storageOptions: {
        skipBackup: true,
        path: 'images',
      },
    };

    /**
     * The first arg is the options object for customization
     * (it can also be null or omitted for default options),
     * The second arg is the callback which sends object: response (more info below in README)
     */
    ImagePicker.showImagePicker(options, (response) => {
      console.log('Response = ', response);

      if (response.didCancel) {
        console.log('User cancelled image picker');
      } else if (response.error) {
        console.log('ImagePicker Error: ', response.error);
      } else if (response.customButton) {
        console.log('User tapped custom button: ', response.customButton);
      } else {
        const source = response.uri;

        // You can also display the image using data:
        // let source = { uri: 'data:image/jpeg;base64,' + response.data };

        this.setState({
          avatarSource: source,
        });
      }
    });
  }

  render() {
    const pic = this.state.avatarSource || this.props.value;

    return (
      <View style={formStyles.imagePickerWrapper}>
        {
          pic ?
            <Image
              source={{ uri: pic }}
              style={formStyles.imagePickerImage}
            />
            :
            null
        }
        <TouchableOpacity onPress={() => this.showPicker()}>
          <Text style={formStyles.imagePickerReplaceText}>Select image</Text>
        </TouchableOpacity>
      </View>
    );
  }
}
