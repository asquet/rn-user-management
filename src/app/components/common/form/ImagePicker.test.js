import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import ImagePicker from './ImagePicker';

// Note: test renderer must be required after react-native.

describe('ImagePicker', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <ImagePicker />,
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
