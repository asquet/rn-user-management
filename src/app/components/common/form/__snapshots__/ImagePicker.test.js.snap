// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImagePicker renders correctly 1`] = `
<View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Select image
    </Text>
  </View>
</View>
`;
